presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -101,6 +101,7 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
             Estimate nullsFraction;
             if (hiveColumnHandle.isPartitionKey()) {
                 rangeStatistics.setDistinctValuesCount(countDistinctPartitionKeys(hiveColumnHandle, hivePartitions));
+                rangeStatistics.setDataSize(calculateDataSize(partitionStatistics, columnName));
```

Comment:
This is not correct. You have to compute it based on the `HivePartition#keys`. `partitionStatistics` don't contain any information about the partition columns. Only data columns are there.

Reply:
i see.  I misunderstood what @findepi meant.

Reply:
@rschlussel2 my bad, i totally forgot. I had a PR in the queue for this. Let me rebase this code after recent changes here and see how it goes

Reply:
https://github.com/prestodb/presto/pull/11107